app/src/main/java/com/kabouzeid/gramophone/util/PlaylistsUtil.java

```diff
@@ -168,6 +187,17 @@ public static void removeFromPlaylist(@NonNull final Context context, @NonNull f
         }
     }
 
+    public static void removeFromPlaylist(@NonNull final Context context, @NonNull final Song song, int playlistId, int startID) {
```

Comment:
Could you explain what the start ID is for?

Reply:
startID gets the value of base in case a insertion of a duplicate should be reverted. If you just call the function with a Song **every** occurence of the song will be deleted. The startID can be passed as the base of the insertion so it is guaranteed that only the recently added song, not the one already inside the playlist will be deleted. (MediaStore.Audio.Playlists.Members.PLAY_ORDER + " >=?"   is the query to also find the recently added song in the delete-call)
This could be made the standard-parameter-set for removeFromPlaylist, startID could be 0 to do the same as it does currently without the startID. But it would increase the overhead in the classic deletion process.

Reply:
Thinking about the deletion, it might actually be possible that if you delete a song afterwards all songs with that id will be deleted from the playlist, I will have to test this.